sympy/series/formal.py

```diff
@@ -0,0 +1,445 @@
+    to increase the range of rational algorithm. See :func:`rational_algorithm`
+    for details.
+
+    See Also
+    ========
+
+    sympy.series.rational_algorithm
+    """
+    if x0 in [S.Infinity, -S.Infinity]:
+        dir = {S.Infinity: S.One, S.NegativeInfinity: -S.One}[x0]
+        temp = f.subs(x, 1/x)
+        result = compute_fps(temp, x, 0, dir, hyper, order, rational, full)
+        if result is None:
+            return None
+        return result[0], result[1].subs(x, 1/x), result[2].subs(x, 1/x)
+    elif x0 or dir == -S.One:
+        if dir == -S.One:
+            rep = -x + x0
+            rep2 = -x
+            rep2b = x0
+        else:
+            rep = x + x0
+            rep2 = x
+            rep2b = -x0
+        temp = f.subs(x, rep)
+        result = compute_fps(temp, x, 0, S.One, hyper, order, rational, full)
+        if result is None:
+            return None
+        return (result[0], result[1].subs(x, rep2 + rep2b),
+                result[2].subs(x, rep2 + rep2b))
+
+    result = None
+
+    # from here on it's x0=0 and dir=1 handling
+    if rational:
+        k = Dummy('k')
+        result = rational_algorithm(f, x, k, order, full)
+
+    if result is None and hyper:
+        return None # TODO
+
+    if result is None:
+        return None
+
+    ak = sequence(result[0], (k, result[2], oo))
+    xk = sequence(x**k, (k, 0, oo))
+    ind = result[1]
+
+    return ak, xk, ind
+
+
+class FormalPowerSeries(SeriesBase):
+    """Represents Formal Power Series
+
+    No computation is performed.
+    This class should only to be used to represent
+    a series. No checks are performed.
+
+    For computing a series use :func:`fps`.
+
+    See Also
+    ========
+
+    sympy.series.formal.fps
+    sympy.series.formal.compute_fps
+    """
+    def __new__(cls, *args):
+        args = map(sympify, args)
+        return Expr.__new__(cls, *args)
+
+    @property
+    def function(self):
+        return self.args[0]
+
+    @property
+    def x(self):
+        return self.args[1]
+
+    @property
+    def x0(self):
+        return self.args[2]
+
+    @property
+    def ak(self):
+        return self.args[4][0]
+
+    @property
+    def xk(self):
+        return self.args[4][1]
+
+    @property
+    def ind(self):
+        return self.args[4][2]
+
+    @property
+    def interval(self):
+        return Interval(0, oo)
+
+    @property
+    def start(self):
+        return self.interval.inf
+
+    @property
+    def stop(self):
+        return self.interval.sup
+
+    @property
+    def length(self):
+        return oo
+
+    @property
+    def infinite(self):
+        """returns an infinite representation of the series"""
+        from sympy.concrete import Sum
+
+        ak, xk = self.ak, self.xk
+        k = ak.variables[0]
+
+        return self.ind + Sum(ak.formula * xk.formula, (k, ak.start, ak.stop))
+
+    def polynomial(self, n=6):
+        """truncated series as polynomial.
```

Comment:
Capitalize. You do this in a couple more places, not going to bother tagging all of them. Please proofread your docstrings.